pgx/Cargo.toml

```diff
@@ -33,26 +33,33 @@ no-default-features = true
 rustc-args = ["--cfg", "docsrs"]
 
 [dependencies]
-bitflags = "1.3.2"
-bitvec = "1.0" # processing array nullbitmaps
-cstr_core = "0.2.5"
 enum-primitive-derive = "0.2.2"
 num-traits = "0.2.15"
-seahash = "4.1.0"
-libc = "0.2.126"
+
 pgx-macros = { path = "../pgx-macros/", version = "=0.5.0-beta.0" }
 pgx-pg-sys = { path = "../pgx-pg-sys", version = "=0.5.0-beta.0" }
 pgx-utils = { path = "../pgx-utils/", version = "=0.5.0-beta.0" }
-serde = { version = "1.0.137", features = [ "derive" ] }
-serde_cbor = "0.11.2"
-serde_json = "1.0.81"
-time = { version = "0.3.9", features = ["formatting", "parsing", "alloc", "macros"] } # TODO(0.6.0): add `optional = true`
-atomic-traits = "0.3.0"
-heapless = "0.7.13"
-uuid = { version = "1.0.0", features = [ "v4" ] }
-once_cell = "1.10.0"
+
+# used to internally impl things
+once_cell = "1.10.0" # polyfill until std::lazy::OnceCell stabilizes
+quote = "1.0.18" # necessary for pgx's SQL generation
+uuid = { version = "1.0.0", features = [ "v4" ] } # PgLwLock and shmem
+
+# error handling and logging
 eyre = "0.6.8"
+thiserror = "1.0"
 tracing = "0.1.34"
 tracing-error = "0.2.0"
-quote = "1.0.18"
-thiserror = "1.0"
+
+# exposed in public API
```

Comment:
fwiw I considered something "exposed" if it's "encapsulated but basically we've documented it definitely uses this, so we're stuck using it or completely reimplementing it or something similar", which is where seahash is at, but most of these aren't even that, they're just Right There.